test/silhouette.jl

```diff
@@ -34,4 +34,21 @@ c = [3, 1]
 
 @test silhouettes(a, c, D) ≈ [0.5, 0.5, -1/3, 0.0]
 
+@testset "silhouette() errors when degenerated clustering is given" begin
+    a = [fill(1, 5); fill(2, 5)]
+    d = fill(0, (10, 10))
+
+    @test silhouettes(a, d) ≈ fill(0.0, 10)
+
+    d = fill(1, (10, 10))
+    d[1, 2] = d[2, 1] = 5
+    
+    @test silhouettes(a, d) ≈ [[-0.5, -0.5]; fill(0.0, 8)]
+
+    a = fill(1, 10)
```

Comment:
That's the second testset: `@testset "silhouette() throws an error when degenerated clustering is given" begin`